scripts/usb_reset.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; version 2.1 only. #
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# ./usb_reset.py 1-2
+# reset USB device 1-2
+
+import fcntl
+import xcp.logger as log
+import logging
+import os
+import re
+import sys
+import traceback
+
+#USBDEVFS_RESET _IO('U', 20)
+USBDEVFS_RESET = (ord('U') << 8) | 20
+
+
+def read_int(path):
+    with open(path) as f:
+        return int(f.readline())
+
+
+def reset_device(bus, dev):
+    fd = os.open("/dev/bus/usb/{0:03d}/{1:03d}".format(bus, dev), os.O_WRONLY)
```

Comment:
Can you not open a `file` object and then use `fileno()` to get the fd?

Reply:
Yes, I can.